gridded_etl_tools/utils/zarr_methods.py

```diff
@@ -827,15 +867,26 @@ def calculate_update_time_ranges(
         # Calculate a tuple of the start/end indices for each datetime range
         regions_indices = []
         for date_pair in datetime_ranges:
-            start_int = list(original_dataset.time.values).index(
-                original_dataset.sel(time=date_pair[0], method="nearest").time
-            )
-            end_int = (
-                list(original_dataset.time.values).index(
-                    original_dataset.sel(time=date_pair[1], method="nearest").time
+            if not self.forecast:
+                start_int = list(original_dataset[self.time_dim].values).index(
+                    original_dataset.sel(time=date_pair[0], method="nearest").time
+                )
+                end_int = (
+                    list(original_dataset[self.time_dim].values).index(
+                        original_dataset.sel(time=date_pair[1], method="nearest").time
+                    )
+                    + 1
+                )
+            elif self.forecast:
+                start_int = list(original_dataset[self.time_dim].values).index(
+                    original_dataset.sel(forecast_reference_time=date_pair[0], method="nearest").time
+                )
+                end_int = (
+                    list(original_dataset[self.time_dim].values).index(
+                        original_dataset.sel(forecast_reference_time=date_pair[1], method="nearest").time
```

Comment:
How does the `.time` selection work? Isn't the time dimension called `forecast_reference_time` for forecast datasets?

Reply:
Good catch, will fix this.